osbuild/util/mnt.py

```diff
@@ -44,14 +47,14 @@ class MountGuard(contextlib.AbstractContextManager):
     def __init__(self):
         self.mounts = []
 
-    def mount(self, source, target, bind=True, ro=False, rw=False, mode="0755"):
+    def mount(self, source, target, bind=True, permissions=None, mode="0755"):
         options = []
         if bind:
             options += ["bind"]
-        if ro:
-            options += ["ro"]
-        if rw:
-            options += ["rw"]
+        if permissions:
+            if permissions not in [READ_ONLY_MODE, READ_WRITE_MODE]:
+                raise ValueError(f"unknown filesystem permissions: {permissions}")
+            options += [permissions]
```

Comment:
This would lead to `[[READ_ONLY_MODE, READ_WRITE_MODE]]` as `permissions` is a list itself.

Reply:
A single string. Open to ideas for a better name 😅